compiler/rustc_expand/src/mbe/quoted.rs

```diff
@@ -221,6 +233,15 @@ fn parse_tree(
                     let (ident, is_raw) = token.ident().unwrap();
                     let span = ident.span.with_lo(span.lo());
                     if ident.name == kw::Crate && !is_raw {
+                        // If we use `ident.span` here, $crate will refer to the crate where
+                        // the crate token appears; we want it to refer to the crate which
+                        // is actually defining this macro (the current local crate). This is
+                        // most easily accomplished with a macro expanded macro_rules! which
+                        // uses `$$crate` to get a `$crate` in the expanded macro definition.
+                        let mut expn = span.ctxt().outer_expn_data();
+                        expn.parent = ExpnId::root();
+                        let span =
+                            span.with_call_site_ctxt(LocalExpnId::fresh(expn, ctx).to_expn_id());
```

Comment:
Note to self: push the local change you made while testing to use `Span::fresh_expansion` rather than `with_call_site_ctxt`